lib/vf_iterators.cpp

```diff
@@ -53,18 +53,18 @@ namespace ValueFlow
                 continue;
             if (!astIsContainer(tok))
                 continue;
-            Token* ftok = nullptr;
-            const Library::Container::Yield yield = findIteratorYield(tok, const_cast<const Token**>(&ftok), settings);
-            if (ftok) {
-                Value v(0);
-                v.setKnown();
-                if (yield == Library::Container::Yield::START_ITERATOR) {
-                    v.valueType = Value::ValueType::ITERATOR_START;
-                    setTokenValue(ftok->next(), std::move(v), settings);
-                } else if (yield == Library::Container::Yield::END_ITERATOR) {
-                    v.valueType = Value::ValueType::ITERATOR_END;
-                    setTokenValue(ftok->next(), std::move(v), settings);
-                }
+            const Token* ftok = nullptr;
+            const Library::Container::Yield yield = findIteratorYield(tok, ftok, settings);
+            if (!ftok)
+                continue;
+            Value v(0);
```

Comment:
The obvious partially unnecessary creation of the object will be addressed in a follow-up PR which contains more such cases in the split out ValueFlow code. I want to profile those changes in detail to see if it might have an impact.